plugins/esi/combo_handler.cc

```diff
@@ -136,6 +138,24 @@ struct InterceptData {
   ~InterceptData();
 };
 
+/*
+ * This class is responsible for keeping track of and processing the various
+ * Cache-Control values between all the requested documents
+ */
+struct CacheControlHeader {
+  // Update the object with a document's Cache-Control header
+  void update(TSMBuffer bufp, TSMLoc hdr_loc);
+
+  // Return the Cache-Control for the combined document
+  string generate() const;
+
+  // Cache-Control values we're keeping track of
+  int _max_age    = 315360000; // max value (10 years)
+  bool _public    = true;
+  bool _private   = false;
```

Comment:
Yeah the tri-enum sounds better. I knew that public & private were disjoint, but there was the default case I wanted to handle too.